src/main/scala/com/atomist/tree/content/text/AbstractMutableContainerTreeNode.scala

```diff
@@ -81,23 +71,30 @@ abstract class AbstractMutableContainerTreeNode(val nodeName: String)
       fv <- _fieldValues
     } {
       fv match {
-        case sm: MutableTerminalTreeNode if sm.startPosition != null && sm.startPosition.offset >= 0 && sm.startPosition.offset >= lastEndOffset =>
-          val smoffset = sm.startPosition.offset
-          if (smoffset > lastEndOffset) fieldResults.append(padding(lastEndOffset, smoffset))
-          lastEndOffset = sm.endPosition.offset
-          fieldResults.append(sm)
-        case soo: AbstractMutableContainerTreeNode =>
-          try {
-            soo.pad(initialSource)
+        case sm: PositionedTreeNode if sm.startPosition != null && sm.startPosition.offset >= 0 =>
+          // This condition isn't pretty, is it? No. I suspect we are not testing the right conditions.
+          // Perhaps we mean "if this node isn't a bunch of whitespace" (that's a thing in Python)
```

Comment:
This looks to me like this should be moved into a meaningfully named method on `PositionedTreeNode`